jpyinterpreter-docs/src/modules/ROOT/pages/jpyinterpreter-architecture/jpyinterpreter-architecture.adoc

```diff
@@ -0,0 +1,125 @@
+[[jpyinterpreterArchitecture]]
+= JPyInterpreter Architecture
+
+The architecture of JPyInterpreter is composed of several components spread across Python and Java.
+The Python components are:
+
+- `jvm_setup.py`, which sets up the JVM and the hooks JPyInterpreter uses to communicate with CPython (to look up packages, call native methods, and converting CPython object to Java object and vice-versa).
+
+- `python_to_java_bytecode_translator.py`, which acts as the Python's frontend to JPyInterpreter.
+Users supply a CPython function to translate (and an optional Java functional interface it implements) to `translate_python_bytecode_to_java_bytecode`, which then converts that function to a `PythonCompiledFunction` and pass it to `PythonBytecodeToJavaBytecodeTranslator` to translate the function.
```

Comment:
```suggestion
Users supply a CPython function to translate (and an optional Java functional interface it implements) to `translate_python_bytecode_to_java_bytecode`, which then converts that function to a `PythonCompiledFunction` and passes it to `PythonBytecodeToJavaBytecodeTranslator` to translate the function.
```

Depends on who actually passes what to where. Maybe a well-placed comma would clarify.